### PR TITLE
Fix custom region logic

### PR DESCRIPTION
**Pull request by Southclaws**
Previously, this code used Rusoto to get the default region as an object then, if there was a custom endpoint turn that region into a Region::Custom with the previously acquired default region name as the custom name. However, Region::default() performs internal checks to filter out names that do not match the official AWS region names and defaults to 'us-east-1' if it's not in the list. This result in custom region names always being ignored and defaulting to 'us-east-1'. The fix changes the logic around this to first check if a custom endpoint is used and then check for custom region names - if no custom endpoint is specified, it simply uses the AWS default.

### CODE REVIEW

**Comment by Southclaws** [comment]
This was the issue I ran into in #13 - turns out the region was always defaulting to `us-east-1`.

File: src/storage/s3.rs [hunk 1]
```diff
@@ -105,15 +105,18 @@ impl Backend {
...
-        if let Ok(endpoint) = std::env::var("AWS_S3_ENDPOINT") {
-            region = Region::Custom {
-                name: region.name().to_owned(),
+        // If a custom endpoint is set, do not use the AWS default (us-east-1)
+        // instead, check environment variables for a region name.
+        let region = if let Ok(endpoint) = std::env::var("AWS_S3_ENDPOINT") {
+            Region::Custom {
+                name: std::env::var("AWS_DEFAULT_REGION")
+                    .or_else(|_| std::env::var("AWS_REGION"))
+                    .unwrap(),
```
Comment (jasonwhite) [inline src/storage/s3.rs]: Is there anything we can do about this `.unwrap()`? If neither `AWS_DEFAULT_REGION` nor `AWS_REGION` are set, shouldn't it fall back to `us-east-1`? Something like this perhaps?
```rust
std::env::var("AWS_DEFAULT_REGION")
    .or_else(|_| std::env::var("AWS_REGION"))
    .unwrap_or_else(|| String::from("us-east-1")),
```

Comment (Southclaws) [reply src/storage/s3.rs]: I had a feeling you'd pick on that!

So I was going to do a default but it felt like it makes no sense to default to `us-east-1` when that region isn't going to exist on a custom S3 service. For example, Scaleway uses regions named `fr-par` and `nl-ams`, Digital Ocean uses `nyc3`, `fra1` etc. So it felt better to throw an error here.

Now, I must admit I couldn't figure out how to actually _return_ a Future that yields an error - I'm super rusty with the Futures API!

**Comment by jasonwhite** [comment]
Merged this so it doesn't sit here and rot forever.

**Pull request merged**